src/pcp/js/widgets.js

```diff
@@ -0,0 +1,304 @@
+                method: "attr",
+                args: ["{that}.options.attrs"]
+            }
+        },
+        renderOnInit: true
+    });
+
+    fluid.defaults("gpii.pcp.widgets.button", {
+        gradeNames: ["fluid.viewComponent"],
+        label: null,
+        selectors: {
+            label: ".fl-btnLabel"
+        },
+        attrs: {
+            // user provided attributes
+        },
+        listeners: {
+            "onCreate.addAttrs": {
+                "this": "{that}.container",
+                method: "attr",
+                args: ["{that}.options.attrs"]
+            },
+            "onCreate.bindClickEvt": {
+                "this": "{that}.container",
+                method: "click",
+                args: ["{that}.onClick"]
+            },
+            "onCreate.initText": {
+                "this": "{that}.dom.label",
+                method: "text",
+                args: ["{that}.options.label"]
+            }
+        },
+        invokers: {
+            onClick: {
+                funcName: "gpii.pcp.noop"
+            }
+        }
+    });
+
+    fluid.defaults("gpii.pcp.widgets.textfield", {
+        gradeNames: ["fluid.viewComponent", "gpii.pcp.widgets.attrsExpander"],
+        selectors: {
+            input: ".flc-textfieldInput"
+        },
+        attrs: {
+            "aria-labelledby": "{that}.model.path"
+        },
+        components: {
+            textfield: {
+                type: "fluid.textfield",
+                container: "{that}.dom.input",
+                options: {
+                    model: "{gpii.pcp.widgets.textfield}.model",
+                    attrs: "{gpii.pcp.widgets.textfield}.options.attrs"
+                }
+            }
+        }
+    });
+
+    fluid.defaults("gpii.pcp.widgets.switch", {
+        gradeNames: ["fluid.switchUI", "gpii.pcp.widgets.attrsExpander"],
+        attrs: {
+            // "aria-labelledby": null
+        },
+        strings: {
+            on: "On",
+            off: "Off"
+        }
+    });
+
+    /**
+     * A function which is executed while the user is dragging the
+     * thumb of a slider.
+     * @param that {Component} An instance of a slider component.
+     * @param container {Object} The jQuery object representing the
+     * slider input.
+     */
+    gpii.pcp.widgets.onSlide = function (that, container) {
+        var value = container.val();
+        that.applier.change("stringValue", value, null, "slide");
+    };
+
+    fluid.defaults("gpii.pcp.widgets.slider", {
+        gradeNames: ["fluid.textfieldSlider"],
+        components: {
+            slider: {
+                options: {
+                    listeners: {
+                        "onCreate.bindSlideEvt": {
+                            "this": "{that}.container",
+                            "method": "on",
+                            "args": ["input", "{that}.onSlide"]
+                        },
+                        "onCreate.bindRangeChangeEvt": {
```

Comment:
I'm not sure the namespace for this event is descriptive - perhaps it is left over from a previous definition?

Reply:
Actually, we are replacing the `onCreate.bindRangeChangeEvt` listener which comes from the default configuration of the `fluid.textfieldSlider` component. We do this in order not to constantly update the slider's model while the user is dragging the thumb. The update occurs when the user lets go of the thumb.